tests/pyspark/optuna/test_study.py

```diff
@@ -89,3 +89,122 @@ def fail_objective(_):
         match="Optimization run for Optuna MlflowSparkStudy failed",
     ):
         mlflow_study.optimize(fail_objective, n_trials=4)
+
+
+def test_auto_resume_existing_study(setup_storage):
+    """Test that MlflowSparkStudy automatically resumes existing studies."""
+    storage = setup_storage
+    study_name = "resume-test-study"
+    sampler = TPESampler(seed=42)
+
+    # Create first study and run some trials
+    study1 = MlflowSparkStudy(study_name, storage, sampler=sampler)
+    assert not study1.is_resumed_study
+
+    def objective(trial):
+        return trial.suggest_float("x", 0, 10) ** 2
+
+    study1.optimize(objective, n_trials=3, n_jobs=1)
+    first_trial_count = len(study1.trials)
+    first_best_value = study1.best_value
+
+    # Create second study with same name - should resume
+    study2 = MlflowSparkStudy(study_name, storage, sampler=sampler)
+    assert study2.is_resumed_study
+    assert len(study2.trials) == first_trial_count
+    assert study2.best_value == first_best_value
+
+    # Continue optimization
+    study2.optimize(objective, n_trials=2, n_jobs=1)
+    assert len(study2.trials) == first_trial_count + 2
```

Comment:
can we also assert that the "resumed study" generates better objective value than the first study ?